visualization/app/codeCharta/ui/dialog/dialog.service.ts

```diff
@@ -31,4 +32,28 @@ export class DialogService {
 
 		return this.$mdDialog.show(prompt)
 	}
+
+	public showValidationWarningDialog(validationResult: CCValidationResult) {
+		const warningSymbol = '<i class="fa fa-exclamation-triangle"></i> '
+
+		const htmlMessage = this.buildHtmlMessage(warningSymbol, validationResult.warning)
+
+		this.showErrorDialog(htmlMessage, validationResult.title)
+	}
+
+	public showValidationErrorDialog(validationResult: CCValidationResult) {
+		const errorSymbol = '<i class="fa fa-exclamation-circle"></i> '
+
+		const htmlMessage = this.buildHtmlMessage(errorSymbol, validationResult.error)
+
+		this.showErrorDialog(htmlMessage, validationResult.title)
+	}
+
+	private buildHtmlMessage(symbol: string, validationResult: string[]): string {
+		const lineBreak = "<br>"
+
+		const errorMessage = validationResult.map(message => symbol + message).join(lineBreak)
+
+		return "<p>" + errorMessage + "</p>"
+	}
```

Comment:
Note: this could be simplified a lot but I do not want to block this PR any longer, so it's nothing to address here. Most of this code is duplicated and it's possible to inline everything into a single function that is probably just five lines long instead of three lines.